File: ENV_estimation/Matlab_env_training/steps_torch_env/FDLP/bandlimit_wtsremoval.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Wed Nov 27 14:56:04 2019

@author: user
"""
import numpy
from pdb import set_trace as bp  #################added break point

#function [new_wts,new_idx] = bandlimit_wtsremoval(wts,idx,flen,sr,minfreq, maxfreq)


def bandlimit_wtsremoval(wts,idx,flen,sr,minfreq,maxfreq):
    cnt=0
    new_idx = []
    new_wts = []
    for I in range (wts.shape[0]):
        tmpx=numpy.zeros((flen,1))
        tmpx[idx[I,0]:idx[I,1]+1]=wts[I]
        
        y=max(tmpx)
        i=numpy.argmax(tmpx)
        #% freq of peak
        f=i*(sr/2)/flen
       
        if f >= minfreq and f<=maxfreq:
            new_idx.append(idx[I,:])
            new_wts.append(wts[I])
            cnt=cnt+1
    new_idx = numpy.asarray(new_idx)
    new_wts = numpy.asarray(new_wts)  
    return new_wts.T,new_idx
    

File: ENV_estimation/Matlab_env_cleaning/matlab_300_MB1_gn_arfit/bandlimit_wtsremoval.m
function [new_wts,new_idx] = bandlimit_wtsremoval(wts,idx,flen,sr,minfreq, maxfreq)

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% function to perform bandlimiting by removing some elements of wts and idx
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

cnt=1;
for I=1:length(wts)
    tmpx=zeros(flen,1);
    tmpx(idx(I,1):idx(I,2))=wts{I};
    
    [y,i]=max(tmpx);
    
    % freq of peak
    f=i*(sr/2)/flen;
    
    % is it inside range
    if (f >= minfreq && f<=maxfreq)
        new_idx(cnt,:)=idx(I,:);
        new_wts{cnt}=wts{I};
        cnt=cnt+1;
    end
end
new_wts=new_wts';
